<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Dr.&nbsp;Panagiotis Chouvardas">

<title>CRC LncRNAs</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="CRC.lcRNAs_files/libs/clipboard/clipboard.min.js"></script>
<script src="CRC.lcRNAs_files/libs/quarto-html/quarto.js"></script>
<script src="CRC.lcRNAs_files/libs/quarto-html/popper.min.js"></script>
<script src="CRC.lcRNAs_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="CRC.lcRNAs_files/libs/quarto-html/anchor.min.js"></script>
<link href="CRC.lcRNAs_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="CRC.lcRNAs_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="CRC.lcRNAs_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="CRC.lcRNAs_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="CRC.lcRNAs_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">CRC LncRNAs</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Dr.&nbsp;Panagiotis Chouvardas </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<p>This document includes all the code behind the study Mihaljevic et al.</p>
<section id="datasets" class="level2">
<h2 class="anchored" data-anchor-id="datasets">Datasets</h2>
<p>ScRNA-seq data were acquired from Lee, HO., Hong, Y., Etlioglu, H.E. et al.&nbsp;Lineage-dependent gene expression programs influence the immune landscape of colorectal cancer. Nat Genet <strong>52</strong>, 594â€“603 (2020). https://doi.org/10.1038/s41588-020-0636-z and the respective GEO dataset <em>GSE144735</em> and <em>GSE132465</em>.</p>
<p>Bulk RNA-seq data were acquired from TCGA, by using GEPIA2 (http://gepia2.cancer-pku.cn/#index).</p>
</section>
<section id="scrna-seq-analysis" class="level2">
<h2 class="anchored" data-anchor-id="scrna-seq-analysis">ScRNA-seq analysis</h2>
<section id="preprocessing" class="level3">
<h3 class="anchored" data-anchor-id="preprocessing">Preprocessing</h3>
<p>We load the necessary libraries.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scMRMA)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pals)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Nebulosa)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggvenn)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DescTools)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">memory.limit</span>(<span class="dv">80000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 80000</code></pre>
</div>
</div>
<p>We read the count matrices and create a Seurat Object with the two datasets. We filter out the cells which have mitochondrial content &gt;5%. We keep all the cells which express at least one features and all the features which are expressed in at least one cell. We generate plots</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>counts1 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"DATA/GSE144735_processed_KUL3_CRC_10X_raw_UMI_count_matrix.txt"</span>, <span class="at">header =</span> T, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>meta1 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"DATA/GSE144735_processed_KUL3_CRC_10X_annotation.txt"</span>, <span class="at">header =</span> T, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(meta1) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-"</span>,<span class="st">"."</span>, <span class="fu">rownames</span>(meta1))</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>obj1 <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> counts1, <span class="at">min.cells =</span> <span class="dv">1</span>, <span class="at">min.features =</span> <span class="dv">1</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>meta1 <span class="ot">=</span> meta1[<span class="fu">colnames</span>(obj1),]</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>obj1<span class="sc">$</span>type <span class="ot">&lt;-</span> meta1<span class="sc">$</span>Class</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>obj1<span class="sc">$</span>Cell_type <span class="ot">&lt;-</span> meta1<span class="sc">$</span>Cell_type</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>obj1[[<span class="st">"percent.mt"</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(obj1, <span class="at">pattern =</span> <span class="st">"^MT-"</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>obj1 <span class="ot">&lt;-</span> <span class="fu">subset</span>(obj1, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">5</span>)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(counts1)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(meta1)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>counts2 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"DATA/GSE132465_GEO_processed_CRC_10X_raw_UMI_count_matrix.txt"</span>, <span class="at">header =</span> T, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>meta2 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"DATA/GSE132465_GEO_processed_CRC_10X_cell_annotation.txt"</span>, <span class="at">header =</span> T, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(meta2) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-"</span>,<span class="st">"."</span>, <span class="fu">rownames</span>(meta2))</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>obj2 <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> counts2,<span class="at">min.cells =</span> <span class="dv">1</span>, <span class="at">min.features =</span> <span class="dv">1</span>)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>meta2 <span class="ot">=</span> meta2[<span class="fu">colnames</span>(obj2),]</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>obj2<span class="sc">$</span>type <span class="ot">&lt;-</span> meta2<span class="sc">$</span>Class</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>obj2<span class="sc">$</span>Cell_type <span class="ot">&lt;-</span> meta2<span class="sc">$</span>Cell_type</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>obj2[[<span class="st">"percent.mt"</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(obj2, <span class="at">pattern =</span> <span class="st">"^MT-"</span>)</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>obj2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(obj2, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">5</span>)</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(counts2)</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(meta2)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a><span class="fu">gc</span>()</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>obj1<span class="sc">$</span>GSE <span class="ot">&lt;-</span> <span class="st">"GSE144735"</span></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>obj2<span class="sc">$</span>GSE <span class="ot">&lt;-</span> <span class="st">"GSE132465"</span></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">merge</span>(obj1, <span class="at">y =</span> obj2)</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(obj, <span class="st">"OUT/merged.Rds"</span>)</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(obj1)</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(obj2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="suppl.-1a" class="level4">
<h4 class="anchored" data-anchor-id="suppl.-1a">Suppl. 1A</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"OUT/merged.Rds"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"nFeature_RNA"</span>, <span class="st">"nCount_RNA"</span>, <span class="st">"percent.mt"</span>), <span class="at">ncol =</span> <span class="dv">3</span>, <span class="at">group.by =</span> <span class="st">"GSE"</span>, <span class="at">cols=</span><span class="fu">kelly</span>()[<span class="fu">c</span>(<span class="dv">14</span>,<span class="dv">11</span>)], <span class="at">pt.size =</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="CRC.lcRNAs_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("OUT/Suppl. 1A. QC.pdf", width = 20, height = 9.9, units = "cm")</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>obj</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat 
28751 features across 49436 samples within 1 assay 
Active assay: RNA (28751 features, 0 variable features)</code></pre>
</div>
</div>
</section>
</section>
<section id="dimensionality-reduction" class="level3">
<h3 class="anchored" data-anchor-id="dimensionality-reduction">Dimensionality reduction</h3>
<p>We normalize and scale the data, identify the most variable features and perform dimensionality reduction.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(obj)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span>  <span class="fu">FindVariableFeatures</span>(obj, <span class="at">selection.method =</span> <span class="st">"vst"</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>all.genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(obj)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(obj, <span class="at">features =</span> all.genes)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(obj, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> obj), <span class="at">npcs =</span> <span class="dv">50</span>)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="CRC.lcRNAs_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="suppl.-1b" class="level4">
<h4 class="anchored" data-anchor-id="suppl.-1b">Suppl. 1B</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">group.by=</span><span class="st">"GSE"</span>, <span class="at">cols=</span><span class="fu">kelly</span>()[<span class="fu">c</span>(<span class="dv">14</span>,<span class="dv">11</span>)], <span class="at">split.by =</span> <span class="st">"GSE"</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"top"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="CRC.lcRNAs_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("OUT/Suppl. 1B. UMAP batch.pdf", width = 20, height = 9.9, units = "cm")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="cell-type-annotation" class="level3">
<h3 class="anchored" data-anchor-id="cell-type-annotation">Cell type annotation</h3>
<p>The original dataset provides the cell type annotations. We aimed to interdependently validate this labeling by re-performing unsupervised cell type annotation (scMRMA + PanglaoDB).</p>
<section id="a" class="level4">
<h4 class="anchored" data-anchor-id="a">1A</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">group.by=</span><span class="st">"Cell_type"</span>, <span class="at">cols=</span><span class="fu">kelly</span>()[<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">7</span>, <span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">8</span>)], <span class="at">label =</span> T, <span class="at">label.box =</span> T, <span class="at">label.color =</span> <span class="st">"white"</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="CRC.lcRNAs_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("OUT/1A. UMAP.cell type.pdf", width = 10, height = 9, units = "cm")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">scMRMA</span>(<span class="at">input=</span>obj, <span class="at">species=</span><span class="st">"Hs"</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>obj[[<span class="st">"cell_type1"</span>]] <span class="ot">&lt;-</span> result<span class="sc">$</span>multiR<span class="sc">$</span>annotationResult[<span class="fu">colnames</span>(obj),<span class="dv">1</span>]</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">melt</span>(<span class="fu">table</span>(obj<span class="sc">$</span>Cell_type, obj<span class="sc">$</span>cell_type1))</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Original"</span>,<span class="st">"Validation"</span>,<span class="st">"Fraction"</span>)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span>Original, <span class="at">y=</span>Fraction, <span class="at">fill=</span>Validation)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>) <span class="sc">+</span> </span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">kelly</span>()[<span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">7</span>, <span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">8</span>)]) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("OUT/Suppl. 1C. cell type validation.pdf", width = 20, height = 9.9, units = "cm")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="cell-type-specificity" class="level3">
<h3 class="anchored" data-anchor-id="cell-type-specificity">Cell type specificity</h3>
<p>We used gencode annotation to identify the lncRNAs and protein coding genes which are measured in the scRNA-seq data. We then compared the cell type specificity between protein coding and lncRNA genes estimating the Gini Coefficient.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>gencode <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"DATA/human_gencode.txt"</span>, <span class="at">header =</span> T)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>lnc <span class="ot">&lt;-</span> <span class="fu">filter</span>(gencode, type <span class="sc">==</span> <span class="st">"lncRNA"</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>lnc <span class="ot">&lt;-</span> lnc[<span class="fu">which</span>(lnc<span class="sc">$</span>SYMBOL <span class="sc">%in%</span> <span class="fu">rownames</span>(obj)),]</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>prot <span class="ot">&lt;-</span> <span class="fu">filter</span>(gencode, type <span class="sc">==</span> <span class="st">"protein_coding"</span>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>prot <span class="ot">&lt;-</span> prot[<span class="fu">which</span>(prot<span class="sc">$</span>SYMBOL <span class="sc">%in%</span> <span class="fu">rownames</span>(obj)),]</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(obj) <span class="ot">&lt;-</span> <span class="st">"Cell_type"</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>lnc_percent <span class="ot">&lt;-</span> <span class="fu">DotPlot</span>(obj, <span class="at">features =</span> <span class="fu">unique</span>(lnc<span class="sc">$</span>SYMBOL))</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>lnc_percent <span class="ot">&lt;-</span> lnc_percent<span class="sc">$</span>data</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>GiniCoef_lnc <span class="ot">&lt;-</span> lnc_percent <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(features.plot) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">gini =</span> <span class="fu">Gini</span>(pct.exp))</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>GiniCoef_lnc<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="st">"lncRNA"</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>prot_percent <span class="ot">&lt;-</span> <span class="fu">DotPlot</span>(obj, <span class="at">features =</span> <span class="fu">unique</span>(prot<span class="sc">$</span>SYMBOL))</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>prot_percent <span class="ot">&lt;-</span> prot_percent<span class="sc">$</span>data</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>GiniCoef_prot <span class="ot">&lt;-</span> prot_percent <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(features.plot) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">gini =</span> <span class="fu">Gini</span>(pct.exp))</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>GiniCoef_prot<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="st">"protein_coding"</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>GiniCoef <span class="ot">&lt;-</span><span class="fu">rbind</span>(GiniCoef_lnc, GiniCoef_prot)</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(<span class="at">object =</span> GiniCoef, <span class="st">"OUT/gini.Rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="b" class="level4">
<h4 class="anchored" data-anchor-id="b">1B</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>GiniCoef <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"OUT/gini.Rds"</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> GiniCoef <span class="sc">%&gt;%</span> <span class="fu">filter</span>(type<span class="sc">==</span><span class="st">"lncRNA"</span>) </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>lnc_n <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(temp<span class="sc">$</span>gini)))</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> GiniCoef <span class="sc">%&gt;%</span> <span class="fu">filter</span>(type<span class="sc">==</span><span class="st">"protein_coding"</span>) </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>prot_n <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(temp<span class="sc">$</span>gini)))</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(GiniCoef, <span class="fu">aes</span>(<span class="at">x=</span>type, <span class="at">y=</span>gini, <span class="at">fill=</span>type)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">stat_compare_means</span>(<span class="at">hjust=</span><span class="dv">0</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">kelly</span>()[<span class="fu">c</span>(<span class="dv">12</span>,<span class="dv">9</span>)]) <span class="sc">+</span> </span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span> </span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="st">"lncRNA"</span>,<span class="st">"protein_coding"</span>), <span class="at">y=</span><span class="fu">c</span>(<span class="fl">0.84</span>, <span class="fl">0.57</span>), <span class="at">label=</span><span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">"(n = "</span>,lnc_n,<span class="st">")"</span>), </span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">paste0</span>(<span class="st">"(n = "</span>,prot_n,<span class="st">")"</span>)), <span class="at">col=</span><span class="st">'black'</span>, <span class="at">size=</span><span class="dv">4</span>, <span class="at">geom=</span><span class="st">"text"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1085 rows containing non-finite values (`stat_boxplot()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1085 rows containing non-finite values
(`stat_compare_means()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="CRC.lcRNAs_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("OUT/1B. cell type.specificity.pdf", width = 10, height = 9, units = "cm")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="epithelial-lncrnas" class="level3">
<h3 class="anchored" data-anchor-id="epithelial-lncrnas">Epithelial lncRNAs</h3>
<p>We performed differential expression to identify the lncRNAs which are significantly (adjusted pvalue &lt; 0.01) enriched in the epithelial cells.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(obj) <span class="ot">&lt;-</span> <span class="st">"Cell_type"</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>gencode <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"DATA/human_gencode.txt"</span>, <span class="at">header =</span> T)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>lnc <span class="ot">&lt;-</span> <span class="fu">filter</span>(gencode, type <span class="sc">==</span> <span class="st">"lncRNA"</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>lnc <span class="ot">&lt;-</span> lnc[<span class="fu">which</span>(lnc<span class="sc">$</span>SYMBOL <span class="sc">%in%</span> <span class="fu">rownames</span>(obj)),]</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>lnc_percent <span class="ot">&lt;-</span> <span class="fu">DotPlot</span>(obj, <span class="at">features =</span> <span class="fu">unique</span>(lnc<span class="sc">$</span>SYMBOL))</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>lnc_percent <span class="ot">&lt;-</span> lnc_percent<span class="sc">$</span>data</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>M <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(obj, <span class="at">ident.1 =</span> <span class="st">"Epithelial cells"</span>, <span class="at">only.pos =</span> T, <span class="at">min.pct =</span> <span class="dv">0</span>, <span class="at">logfc.threshold =</span> <span class="dv">0</span>, <span class="at">features =</span> <span class="fu">unique</span>(lnc<span class="sc">$</span>SYMBOL))</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>M <span class="ot">&lt;-</span> <span class="fu">subset</span>(M, p_val_adj <span class="sc">&lt;</span> <span class="fl">0.01</span>)</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>lnc_percent2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(lnc_percent, features.plot <span class="sc">%in%</span> <span class="fu">rownames</span>(M))</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>lnc_percent2<span class="sc">$</span>id <span class="ot">&lt;-</span> <span class="fu">factor</span>(lnc_percent2<span class="sc">$</span>id, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"B cells"</span>, <span class="st">"Epithelial cells"</span>, <span class="st">"Mast cells"</span>, <span class="st">"Myeloids"</span>, <span class="st">"Stromal cells"</span>, <span class="st">"T cells"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="c" class="level4">
<h4 class="anchored" data-anchor-id="c">1C</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_density</span>(obj, <span class="st">"LINC00523"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="CRC.lcRNAs_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("OUT/1C. Epithelial.lncRNAs.example.pdf", width = 10, height = 9, units = "cm")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="d" class="level4">
<h4 class="anchored" data-anchor-id="d">1D</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(lnc_percent2, <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(id, avg.exp.scaled), <span class="at">y=</span>avg.exp.scaled, <span class="at">fill=</span>id)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">kelly</span>()[<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">7</span>, <span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">8</span>)]) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Expression (Scaled)"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Cell Type"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="CRC.lcRNAs_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("OUT/1D. Epithelial.lncRNAs.expression.pdf", width = 10, height = 9, units = "cm")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
</section>
<section id="integration-with-tcga-data" class="level2">
<h2 class="anchored" data-anchor-id="integration-with-tcga-data">Integration with TCGA data</h2>
<p>We accessed the COAD TCGA data via the GEPIA2 webserver and performed differential expression analysis between tumor and normal samples. Integrating these results with the epithelial lncRNAs we identified 77 upregulated epithelial lncRNAs, 11 of which show minimal expression in normal samples.</p>
<section id="a-1" class="level4">
<h4 class="anchored" data-anchor-id="a-1">2A</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>COAD <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"DATA/coad.tcga.txt"</span>, <span class="at">header =</span> T)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>lnc <span class="ot">&lt;-</span> <span class="fu">filter</span>(gencode, type <span class="sc">==</span> <span class="st">"lncRNA"</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>COAD <span class="ot">&lt;-</span> <span class="fu">filter</span>(COAD, Gene <span class="sc">%in%</span> lnc<span class="sc">$</span>SYMBOL)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">list</span>(COAD<span class="sc">$</span>Gene[<span class="fu">which</span>(COAD<span class="sc">$</span>log2FC <span class="sc">&gt;</span><span class="dv">0</span>)], COAD<span class="sc">$</span>Gene[<span class="fu">which</span>(COAD<span class="sc">$</span>log2FC <span class="sc">&lt;</span> <span class="dv">0</span>)], <span class="fu">rownames</span>(M))</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(x) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"TCGA-Up"</span>,<span class="st">"TCGA-Down"</span>, <span class="st">"Epithelial"</span>)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggvenn</span>(</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  x, </span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">fill_color =</span> <span class="fu">c</span>(<span class="fu">viridis</span>(<span class="dv">11</span>)[<span class="fu">c</span>(<span class="dv">11</span>,<span class="dv">1</span>)], <span class="fu">kelly</span>()[<span class="fu">c</span>(<span class="dv">7</span>)]),<span class="at">show_percentage =</span> F, </span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">stroke_size =</span> <span class="fl">0.5</span>, <span class="at">set_name_size =</span> <span class="dv">4</span>, <span class="at">fill_alpha=</span><span class="fl">0.85</span>,</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"TCGA - Epithelial lncRNAs overlap"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">face=</span><span class="st">"bold"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="CRC.lcRNAs_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("OUT/2A. Epithelial.lncRNAs.TCGA.venn.pdf", width = 10, height = 9, units = "cm")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="b-1" class="level4">
<h4 class="anchored" data-anchor-id="b-1">2B</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>COAD <span class="ot">&lt;-</span> <span class="fu">filter</span>(COAD, Gene <span class="sc">%in%</span> <span class="fu">rownames</span>(M))</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(COAD, <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(Gene, log2FC), <span class="at">y=</span>log2FC, <span class="at">fill=</span>log2FC)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">midpoint =</span> <span class="dv">0</span>, <span class="at">low =</span> <span class="fu">viridis</span>(<span class="dv">3</span>)[<span class="dv">1</span>], <span class="at">high =</span> <span class="fu">viridis</span>(<span class="dv">3</span>)[<span class="dv">3</span>]) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"LncRNAs"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="CRC.lcRNAs_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("OUT/2B. Epithelial.lncRNAs.TCGA.DEGs.pdf", width = 10, height = 20, units = "cm")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="c-1" class="level4">
<h4 class="anchored" data-anchor-id="c-1">2C</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(COAD, <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">log</span>(Tumor<span class="fl">+0.001</span>), <span class="at">y=</span><span class="fu">log</span>(Normal<span class="fl">+0.001</span>))) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"log(Normal Expression)"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"log(Tumor Expression)"</span>) <span class="sc">+</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">nudge_y =</span> <span class="dv">3</span>,</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">nudge_x =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="fu">kelly</span>()[<span class="dv">7</span>], <span class="at">fill=</span><span class="st">"white"</span>,</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">max.overlaps =</span> <span class="dv">100000</span>, <span class="at">size=</span><span class="dv">2</span>,</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">label=</span><span class="fu">ifelse</span>(COAD<span class="sc">$</span>Normal <span class="sc">==</span> <span class="dv">0</span>, COAD<span class="sc">$</span>Gene, <span class="st">""</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="CRC.lcRNAs_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("OUT/2C. Epithelial.lncRNAs.TCGA.ZeroNormal.pdf", width = 10, height = 9, units = "cm")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>